cacheops/redis.py

```diff
@@ -29,24 +30,48 @@ def handle_connection_failure(call):
 
 client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
 client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
+redis_replicas = []
+
+
+def ip(url):
+    return socket.gethostbyname(urlparse(url).hostname)
+
+
+def set_redis_replicas():
+    # the conf could be a list or string
+    # list would look like: ["redis://cache-001:6379/1", "redis://cache-002:6379/2"]
+    # string would be: "redis://cache-001:6379/1,redis://cache-002:6379/2"
+    global redis_replicas
+    if isinstance(settings.REDIS_REPLICAS, six.string_types):
+        temp = settings.REDIS_REPLICAS.split(',')
+    else:
+        temp = list(settings.REDIS_REPLICAS)
+    master_url = settings.REDIS_MASTER
+    master_ip = ip(master_url)
+    temp = [r for r in temp if ip(r) != master_ip]
+    temp = [r for r in temp for _ in range(settings.REDIS_REPLICA_WEIGHT)]
```

Comment:
Why not add a required # of replica "doubles" directly to `settings.REDIS_REPLICAS`?
This way:
1. We'll be able to assign different weight to different slaves if necessary
2. Will avoid this extra iteration
3. Anyways, number of duplicates is not something computed but set through the config.
Thoughts?

Reply:
The whole point of this is that the primary's `CNAME` will change IPs on failover – one of the hostnames in the `REDIS_REPLICAS` list is actually the primary, but we don't know which until this code runs. So 3 is false, the number of duplicates of each hostname _is_ computed, and we can't do this in the config.